src/controllers/companyController.js

```diff
@@ -0,0 +1,87 @@
+import { companies } from "../models/index.js";
+import notFoundError from "../erros/notFoundError.js";
+
+class CompanyController {
+  static async registerCompany(req, res, next) {
+    try {
+      const newCompany = await companies.create(req.body);
```

Comment:
Que tal trazer as validações dos dados da company antes de salvar os dados da Company para cá?

Eu percebi que você criou as validações direto no schema do Mongoose, mas por mais que exista esse recurso de validação no próprio schema, não é uma boa prática deixar apenas o schema responsável pela validação, por alguns motivos:

- Esse tipo de validação é uma regra de negócio, ou seja, é específica para o contexto descrito na task. Podem haver outros contextos onde eu possa salvar uma company sem nome, ou sem cidade, ou sem documento, e deixar essas validações no schema acaba amarrando teu código para só aceitar esse tipo de configuração.

- Caso um dia você decida mudar de banco de dados, talvez não seja tão simples de validar os dados como é feito no schema do mongoose, então acaba sendo melhor que o banco de dados seja 'burro' para validação dos dados. O ideal é que a única coisa que o schema faça é definir é o contrato de tipos e estrutura do objeto, as demais validações deixamos como regra de negócio.

- Fazer a validação antes de persistir os dados te poupa um pouco de tempo, pois tu não precisa interagir com o banco para validar os dados, isso é feito praticamente de forma instantânea a partir do momento em que a requisição é feita. Nesse contexto dessa API não faz taaaanta diferença, mas em contextos mais complexos é mais eficiente que a validação de formulários seja a primeira coisa a ser feita antes dos demais tratamentos.

- Não é o caso dessa API, mas vamos supor que antes de salvar os dados da Company tu precisasse fazer uma consulta no site da Receita Federal só para validar se o CNPJ é de uma empresa que está ativa. Se a validação do CNPJ ser válido não for feita antes de enviar a requisição para o site da Receita Federal, tu vai acabar gerando uma redundância de validação, pois o site da Receita provavelmente vai retornar o mesmo erro de validação que já poderia ter sido feita na API.

Reply:
CÔSA LINDA LUCAO!

Já estou fazendo!